File: stack.py
class Stack:
    def __init__(self):
        self.stack = []

    def __str__(self):
        return str(self.stack)

    def push(self, value):
        self.stack.append(value)

    def pop(self):
        if not self.stack:
            raise IndexError("Pop from an empty stack")
        index = len(self.stack)-1
        top_element = self.stack[index]
        self.stack = self.stack[:index]
        return top_element
    

s = Stack()
s.push(10)
print(s)
s.push(20)
print(s)
assert (s.pop() == 20)
print(s)

File: stack.c
 
#include <stdio.h>



#define MAX_SIZE 100

typedef struct {
    int data[MAX_SIZE];
    int top;
} Stack;

void init(Stack *s){
    s->top = -1;
}

int push(Stack *s, int value){
    if (s->top == MAX_SIZE-1){
        return -1;
    }
    s->top ++;
    s->data[s->top] = value;
    return 0;

}

int pop(Stack *s, int *popped_value){
    if (s->top == -1){
        return -1;
    }
    *popped_value = s->data[s->top];
    s->top --;

}

int main(){
    Stack s;
    init(&s);

    push(&s, 10);
    push(&s, 20);

    int value;
    pop(&s, &value);
    printf("Popped: %d\n", value);
    pop(&s, &value);
    printf("Popped: %d\n", value);

    if(pop(&s, &value) == -1){
        printf("Underflow handled\n");
    }
    return 0;
}